library/src/main/java/com/codetroopers/betterpickers/recurrencepicker/EventRecurrenceFormatter.java

```diff
@@ -110,7 +110,9 @@ public static String getRepeatString(Context context, Resources r, EventRecurren
                     if(dayNumber == RecurrencePickerDialogFragment.LAST_NTH_DAY_OF_WEEK){
                         dayNumber = 5;
                     }
-                    details = mMonthRepeatByDayOfWeekStrs[weekday][dayNumber - 1];
+                    if (dayNumber > 0) {
```

Comment:
This just prevents the exception - we would have a "monthly > on the same day every month" return a "monthly > on the {n}th {weekday} of the month" string.